app/main.py

```diff
@@ -1,34 +1,65 @@
+from typing import List, Tuple, Optional
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row: int = row
+        self.column: int = column
+        self.is_alive: bool = is_alive
+
+    def hit(self) -> None:
+        self.is_alive = False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: Tuple[int, int], end: Tuple[int, int]) -> None:
+        self.decks: List[Deck] = []
+        self.is_drowned: bool = False
+
+        if start[0] == end[0]:
+            for y_point in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], y_point))
+        else:
+            for x_point in range(start[0], end[0] + 1):
+                self.decks.append(Deck(x_point, start[1]))
```

Comment:
The logic for determining if a ship is horizontal or vertical is correct, but it could be improved for clarity. Consider adding a check to ensure that either the rows or columns are the same for the start and end points, as this would make the code more robust and easier to understand.